test/sv/interfaces.mlir

```diff
@@ -0,0 +1,40 @@
+// RUN: circt-opt %s | FileCheck %s
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+// CHECK-LABEL: func @test() {
+func @test() {
+  // Basic interface smoke test
+
+  sv.interface @myinterface {
+    sv.interface.signal @data : i32
+    sv.interface.modport @input_port ("input" @data)
+    sv.interface.modport @output_port ("output" @data)
+  }
+
+  // CHECK-NEXT: sv.interface @myinterface {
+  // CHECK-NEXT:   sv.interface.signal @data : i32
+  // CHECK-NEXT:   sv.interface.modport @input_port ("input" @data)
+  // CHECK-NEXT:   sv.interface.modport @output_port ("output" @data)
+  // CHECK-NEXT: }
+
+  // Hanshake-like interface smoke test
+
+  sv.interface @handshake_example {
+    sv.interface.signal @data : i32
+    sv.interface.signal @valid : i1
+    sv.interface.signal @ready : i1
+    sv.interface.modport @dataflow_in ("input" @data, "input" @valid, "output" @ready)
+    sv.interface.modport @dataflow_out ("output" @data, "output" @valid, "input" @ready)
+  }
+
+  // CHECK-NEXT: sv.interface @handshake_example {
+  // CHECK-NEXT:   sv.interface.signal @data : i32
+  // CHECK-NEXT:   sv.interface.signal @valid : i1
+  // CHECK-NEXT:   sv.interface.signal @ready : i1
+  // CHECK-NEXT:   sv.interface.modport @dataflow_in ("input" @data, "input" @valid, "output" @ready)
+  // CHECK-NEXT:   sv.interface.modport @dataflow_out ("output" @data, "output" @valid, "input" @ready)
+  // CHECK-NEXT: }
+
```

Comment:
Should probably also add a test using an interface modport as a module parameter

Reply:
This would involve writing a sv.interface type though (?), so maybe not. OTOH this patch feels somewhat incomplete if you're unable to use a modport. (Though I have a use case which doesn't involve this.)

Reply:
Yep, I am already looking into this (using your tblgen support for defining types). In the spirit of submitting incremental changes, I guess I'm hoping this PR has some value on its own, which it sounds like you at least have a use case for.

Besides referring to an interface modport as a module parameter, I think the other big thing that is missing here is EmitVerilog support to actually print the operations out as readable SystemVerilog.

My intent was to try and land the operation definitions while I begin working on the above two items, but if this patch isn't really useful on its own, I can push that work into this branch.

Reply:
I think it's useful on its own. Especially if the next part is pending on my patch landing. I'm planning on merging it into circt/llvm master today.

Reply:
Great! I won't need your patch to begin working on EmitVerilog support, but I'm still thinking it would be best to merge this and have separate PRs for EmitVerilog and actually using an interface as a module parameter.